### ISSUE TITLE
[wayland] Extremely sluggish input

### ISSUE THREAD

**Issue by kennylevinsen** [rxi/lite#72]
Under Wayland (tested with sway), input is very sluggish.

Clicking to move the cursor takes about a wall-lock second to execute. Keyboard input is faster, but not *fast* as tested by moving the cursor up and down with the arrow keys. Also noteworthy is that the cursor bar does not hide when moving the mouse pointer outside the window.

This suggests to me that all of this is caused by unprocessed events from improper event loop handling. This could be a result of polling for more data before the queues are emptied, or incorrect use of prepare_read/read/cancel_read. I have yet to look further into the issue.

To test:
1. Run under wayland, such as through `sway` or GNOME wayland.
2. Set SDL_VIDEODRIVER=wayland.
3. Build a release build of lite, and open it in the repo.
4. Open a random file from the repo, and navigate around.

Expected result:
Same performance and responsiveness as under X11

Actual result:
Extremely sluggish input handling.

**Comment by rxi** [comment]
This sounds like an issue external to lite as it uses SDL for all the input event handling and no setups exhibit these problems. If this does happen to be an issue with SDL, it should be brought up with the SDL team.

**Comment by kennylevinsen** [comment]
Agreed. 99% sure it's SDL, but hadn't seen the issue in other SDL projects, and looked too much at the lite codebase yet.

I'll close this when I confirm the issue.
